### PR TITLE
TD-API: Add the UpdateFederationRelationship method to the datastore

### PR DESCRIPTION
**Pull request by marcosy**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

Datastore

**Description of change**

Add the UpdateFederationRelationship() method to the datastore.

**Which issue this PR fixes**

Fixes: #2506 

Signed-off-by: Marcos Yedro <marcosyedro@gmail.com>

### CODE REVIEW

File: pkg/server/datastore/sqlstore/sqlstore.go [hunk 1]
```diff
@@ -511,6 +512,39 @@ func (ds *Plugin) ListFederationRelationships(ctx context.Context, req *datastor
...
+		case datastore.BundleEndpointWeb:
+		case datastore.BundleEndpointSPIFFE:
+			if fr.EndpointSPIFFEID.IsZero() {
+				return nil, status.Error(codes.InvalidArgument, "bundle endpoint SPIFFE ID is required")
+			}
+		default:
+			return nil, status.Errorf(codes.InvalidArgument, "unknown bundle endpoint profile type: %q", fr.BundleEndpointProfile)
+		}
+	}
+
```
Comment (azdagron) [inline pkg/server/datastore/sqlstore/sqlstore.go:541]: Can this validation be unified with the create method? For example, we could refactor it into a separate function that takes the FederationRelationship and FederationRelationshipMask. The CreateFederationRelationship could pass the "all" mask into the function. What do you think?

**Review by azdagron — APPROVED** [review]
Looks good! Just one suggestion....

File: pkg/server/datastore/sqlstore/sqlstore.go (hunk 1 above)
Comment (marcosy) [reply pkg/server/datastore/sqlstore/sqlstore.go:541]: Sounds great! I'll refactor it.

**Review by azdagron — APPROVED** [review]
\o/ thanks @marcosy !

**Pull request merged**